lib/util/tests/ert_util_normal_path.c

```diff
@@ -0,0 +1,71 @@
+/*
+   Copyright (C) 2017  Statoil ASA, Norway.
+
+   The file 'ert_util_normal_path.c' is part of ERT - Ensemble based Reservoir Tool.
+
+   ERT is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   ERT is distributed in the hope that it will be useful, but WITHOUT ANY
+   WARRANTY; without even the implied warranty of MERCHANTABILITY or
+   FITNESS FOR A PARTICULAR PURPOSE.
+
+   See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+   for more details.
+*/
+
+#include <stdlib.h>
+#include <stdbool.h>
+
+#include <ert/util/test_util.h>
+#include <ert/util/test_work_area.h>
+#include <ert/util/util.h>
+
+
+void test_path(const char * input_path, const char * expected_path) {
+  char * normal_path = util_alloc_normal_path( input_path );
+  test_assert_string_equal( normal_path , expected_path );
+  free( normal_path );
+}
+
+
+void test_relative() {
+  test_work_area_type * work_area = test_work_area_alloc("Work");
+  util_make_path("level0/level1/level2");
+
+
+  test_path( "level0/level1/../", "level0");
```

Comment:
Do we expect this to work at Windows as well? In that case I would appreciate a test with a Windows back slash path as well.

Reply:
> Do we expect this to work at Windows as well?

Ehhh yes but no; we *expect* it to work on Windows, but we do not run the tests on Windows - so how naive is that?!

The way windows path support is implemented is that all operations working with paths use the symbol `UTIL_PATH_SEP_STRING` - and that resolves to `"/"` and `"\"` on Posix and Windows respectively. So - although the code is assumed to be portable to windows - and compile-tested for that - the tests are not written portably.

Reply:
Yeah, that is an interesting approach. But beyond the scope of this PR if it is ever to be fixed.